runtime/compiler/compile/J9SymbolReferenceTable.cpp

```diff
@@ -404,18 +408,37 @@ J9::SymbolReferenceTable::findOrCreateHandleMethodSymbol(TR::ResolvedMethodSymbo
    if (method)
       owningMethodSymbol->setMayHaveInlineableCall(true);
 
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+   TR::SymbolReference * symRef = findOrCreateMethodSymbol(owningMethodSymbol->getResolvedMethodIndex(), cpIndex, method, TR::MethodSymbol::Static);
+#else
    TR::SymbolReference * symRef = findOrCreateMethodSymbol(owningMethodSymbol->getResolvedMethodIndex(), cpIndex, method, TR::MethodSymbol::ComputedVirtual);
+#endif /* J9VM_OPT_OPENJDK_METHODHANDLE */
    return symRef;
    }
 
 
 TR::SymbolReference *
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+J9::SymbolReferenceTable::findOrCreateCallSiteTableEntrySymbol(TR::ResolvedMethodSymbol * owningMethodSymbol, int32_t callSiteIndex, bool isMemberNameObject)
+#else
 J9::SymbolReferenceTable::findOrCreateCallSiteTableEntrySymbol(TR::ResolvedMethodSymbol * owningMethodSymbol, int32_t callSiteIndex)
+#endif
    {
    TR::SymbolReference *symRef;
    TR_SymRefIterator i(aliasBuilder.callSiteTableEntrySymRefs(), self());
    TR_ResolvedMethod *owningMethod = owningMethodSymbol->getResolvedMethod();
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+   J9InvokeCacheEntry *invokeCache = (J9InvokeCacheEntry *) owningMethod->callSiteTableEntryAddress(callSiteIndex);
+   if (!invokeCache)
```

Comment:
`invokeCache` is unlikely to be null based on the way it's calculated. The exception is unnecessary. If it's null, there will be a functional issue and we'll crash trying to access its member variable.